pyface/ui/qt4/widget.py

```diff
@@ -99,13 +102,27 @@ def destroy(self):
             self.control = None
 
     def _add_event_listeners(self):
+        super()._add_event_listeners()
         self.control.installEventFilter(self._event_filter)
 
     def _remove_event_listeners(self):
         if self._event_filter is not None:
             if self.control is not None:
                 self.control.removeEventFilter(self._event_filter)
             self._event_filter = None
+        super()._add_event_listeners()
```

Comment:
```suggestion
        super()._remove_event_listeners()
```

Reply:
A bit worrying that this didn't cause the tests to fail, but the `remove_event_listeners` stuff is very "belt and suspenders" - most of the time it would be fine to just remove the widget.